internal/clientconn/conn.go

```diff
@@ -198,7 +198,43 @@ func (c *conn) run(ctx context.Context) (err error) {
 	for {
 		var reqHeader *wire.MsgHeader
 		var reqBody wire.MsgBody
+		var resHeader *wire.MsgHeader
+		var resBody wire.MsgBody
+		var validationErr *wire.ValidationError
+
 		reqHeader, reqBody, err = wire.ReadMessage(bufr)
+		if err != nil && errors.As(err, &validationErr) {
+			res := new(wire.OpMsg)
+
+			must.NoError(res.SetSections(wire.OpMsgSection{
+				Documents: []*types.Document{validationErr.Document()},
+			}))
+
+			var b []byte
+
+			b, err = res.MarshalBinary()
+			if err != nil {
+				panic(err)
+			}
+
+			resHeader = &wire.MsgHeader{
+				OpCode:        reqHeader.OpCode,
+				RequestID:     atomic.AddInt32(&c.lastRequestID, 1),
+				ResponseTo:    reqHeader.RequestID,
+				MessageLength: int32(wire.MsgHeaderLen + len(b)),
+			}
+
+			if err = wire.WriteMessage(bufw, resHeader, res); err != nil {
+				return
```

Comment:
So, we don't do anything with the error? We don't log it etc? Should we panic or something?

Reply:
Replaced return with panic.